pkg/asset/installconfig/kubevirt/validation_test.go

```diff
@@ -0,0 +1,187 @@
+	validAccessMode       = "valid-access-mode"
+	validMachineCIDR      = "192.168.123.0/24"
+	invalidKubeconfigPath = "invalid-kubeconfig-path"
+	invalidNamespace      = "invalid-namespace"
+	invalidStorageClass   = "invalid-storage-class"
+	invalidNetworkName    = "invalid-network-name"
+	invalidAPIVIP         = "invalid-api-vip"
+	invalidIngressVIP     = "invalid-ingress-vip"
+	invalidAccessMode     = "invalid-access-mode"
+	invalidMachineCIDR    = "10.0.0.0/16"
+)
+
+func validInstallConfig() *types.InstallConfig {
+	return &types.InstallConfig{
+		Networking: &types.Networking{
+			MachineNetwork: []types.MachineNetworkEntry{
+				{CIDR: *ipnet.MustParseCIDR(validMachineCIDR)},
+			},
+		},
+		Platform: types.Platform{
+			Kubevirt: &kubevirt.Platform{
+				Namespace:                  validNamespace,
+				StorageClass:               validStorageClass,
+				NetworkName:                validNetworkName,
+				APIVIP:                     validAPIVIP,
+				IngressVIP:                 validIngressVIP,
+				PersistentVolumeAccessMode: validAccessMode,
+			},
+		},
+	}
+}
+
+func TestKubevirtInstallConfigValidation(t *testing.T) {
+	cases := []struct {
+		name             string
+		edit             func(ic *types.InstallConfig)
+		expectedError    bool
+		expectedErrMsg   string
+		clientBuilderErr error
+		expectClient     func(kubevirtClient *mock.MockClient)
+	}{
+		{
+			name:           "valid",
+			edit:           nil,
+			expectedError:  false,
+			expectedErrMsg: "",
+			expectClient: func(kubevirtClient *mock.MockClient) {
+				kubevirtClient.EXPECT().ListVirtualMachine(gomock.Any()).Return(nil, nil).AnyTimes()
+				kubevirtClient.EXPECT().GetNetworkAttachmentDefinition(gomock.Any(), validNetworkName, validNamespace).Return(nil, nil).AnyTimes()
+				kubevirtClient.EXPECT().GetStorageClass(gomock.Any(), validStorageClass).Return(nil, nil).AnyTimes()
+			},
+		},
+		{
+			name:           "invalid empty platform",
+			edit:           func(ic *types.InstallConfig) { ic.Platform.Kubevirt = nil },
+			expectedError:  true,
+			expectedErrMsg: "platform.kubevirt: Required value: validation requires a Engine platform configuration",
+		},
+		{
+			name:             "invalid client builder error",
+			expectedError:    true,
+			expectedErrMsg:   fmt.Sprintf("platform.kubevirt.InfraClusterReachable: Invalid value: \"InfraCluster\": failed to create InfraCluster client with error: test"),
+			clientBuilderErr: errors.New("test"),
+		},
+		{
+			name:           "invalid cluster unreachable",
+			edit:           nil,
+			expectedError:  true,
+			expectedErrMsg: fmt.Sprintf("platform.kubevirt.InfraClusterReachable: Invalid value: \"InfraCluster\": failed to access to InfraCluster with error: test"),
+			expectClient: func(kubevirtClient *mock.MockClient) {
+				kubevirtClient.EXPECT().ListVirtualMachine(gomock.Any()).Return(nil, fmt.Errorf("test")).AnyTimes()
+			},
+		},
+		{
+			name:           "invalid namespace",
```

Comment:
this test just fails cause clients returns error, not because the namespace is wrong

Reply:
I removed this test case, as it actually duplicate of  "invalid storage class" test case